### PR TITLE
README, spec and migration guide cleanup for react-divider

### PR DESCRIPTION
**Pull request by khmakoto**
## PR Description

This PR cleans up a bunch of things in `react-divider`:
- It updates the migration guide to put it more in line with the guides in other packages.
- The README is updated to point to our storybook docs, reference importing from `react-components`, add usage examples, and point to the spec and migration guides.
- The spec is updated to add links to both the types files and migration guide.

This PR also includes small cleanups in `react-slider` and `react-switch` READMEs.

Fixes part of #23423.

### CODE REVIEW

File: packages/react-components/react-divider/MIGRATION.md [hunk 1]
```diff
@@ -1,60 +1,36 @@
 # Divider Migration
 
-## STATUS: WIP 🚧
-
-This Migration guide is a work in progress and is not yet ready for use.
-
 ## Migration from v8
 
-The existing `Separator` control supports the `content` property as well as children nodes. The `content` property has been removed and now only supports children.
+The existing v8's `Separator` control supports a very similar set of props to the one being proposed for v9's `Divider` with a few differences that are outlined below:
```
Comment (TristanWatanabe) [inline packages/react-components/react-divider/MIGRATION.md:5]:
```suggestion
The existing v8 `Separator` control supports a very similar set of props to the one being proposed for v9's `Divider` with a few differences that are outlined below:
```

File: packages/react-components/react-divider/MIGRATION.md [hunk 2]
```diff
@@ -1,60 +1,36 @@
...
+- `styles` => NOT SUPPORTED - use new styling system via `tokens` instead
+- `theme` => NOT SUPPORTED
+- `vertical` => `vertical`
 
 ## Migration from v0
 
-The v0 Divider is close to the converged Divider. Again, as with the Separator control, the `content` property in the converged component is not supported.
-
-Also, from design discussion, the converged component does not support the `size` component as it is redundant as it can be replicated through using the `borderSize` property and if needed, update the `fontSize` property to match desired style.
+The existing v0's `Divider` control supports a very similar set of props to the one being proposed for v9 with a few differences that are outlined below:
```
Comment (TristanWatanabe) [inline packages/react-components/react-divider/MIGRATION.md:15]:
```suggestion
The existing v0 `Divider` control supports a very similar set of props to the one being proposed for v9 with a few differences that are outlined below:
```

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `184.781 kB`<br />`51.176 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.988 kB`<br />`11.108 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `15.485 kB`<br />`5.534 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |    `25.173 kB`<br />`8.12 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |    `25.37 kB`<br />`8.208 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/bd64779b7d68775260b587f10265d64b916f6140'>bd64779b7d68775260b587f10265d64b916f6140</a></sub>
<div id="bundleSizeComment9423"/>

**Review by TristanWatanabe — APPROVED** [review]

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/Avatar-mount_base.html">1083</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/Avatar-mount.html">1131</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/Button-mount_base.html">721</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/Button-mount.html">661</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProvider-mount_base.html">2310</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProvider-mount.html">2347</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-mount_base.html">389</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-mount.html">364</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">297</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">280</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">396</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">382</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/MakeStyles-mount_base.html">1997</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23439/perf-test-react-components/MakeStyles-mount.html">1893</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-divider/MIGRATION.md (hunk 1 above)
Comment (behowell) [reply packages/react-components/react-divider/MIGRATION.md:5]: Or, since we're editing this sentence, it's probably also better to reword it to remove implications that the v9 Divider doesn't exist yet:
```suggestion
The v8 `Separator` control supports a very similar set of props to the v9 `Divider` with a few differences that are outlined below:
```

File: packages/react-components/react-divider/MIGRATION.md [hunk 3]
```diff
@@ -1,60 +1,36 @@
...
 
-The existing `Separator` control supports the `content` property as well as children nodes. The `content` property has been removed and now only supports children.
+The existing v8's `Separator` control supports a very similar set of props to the one being proposed for v9's `Divider` with a few differences that are outlined below:
 
-With the new converged capabilities, both the `styles` and `theme` properties are no longer supported.
-
-Properties that are still available are the `alignContent` and `vertical` with the same acceptable values that are currently defined.
+- `alignContent` => `alignContent`
+- `content` => NOT SUPPORTED - use `children` instead
+- `styles` => NOT SUPPORTED - use new styling system via `tokens` instead
```
Comment (behowell) [inline packages/react-components/react-divider/MIGRATION.md:9]: Tweaks to wording: `content` is replaced by `children` (but still supported). And `styles` is still a prop, but not recommended.
```suggestion
- `content` => children of the `Divider`
- `styles` => recommended to use new theming system via `FluentProvider`
```

File: packages/react-components/react-divider/MIGRATION.md [hunk 4]
```diff
@@ -1,60 +1,36 @@
...
+- `theme` => NOT SUPPORTED
+- `vertical` => `vertical`
```
Comment (behowell) [inline packages/react-components/react-divider/MIGRATION.md:11]: The sentence above mentions that this is a list of the differences, but it appears `alignContent` and `vertical` are identical. Should they be left out of this list?

File: packages/react-components/react-divider/MIGRATION.md [hunk 5]
```diff
@@ -1,60 +1,36 @@
...
 
 ## Migration from v0
 
-The v0 Divider is close to the converged Divider. Again, as with the Separator control, the `content` property in the converged component is not supported.
-
-Also, from design discussion, the converged component does not support the `size` component as it is redundant as it can be replicated through using the `borderSize` property and if needed, update the `fontSize` property to match desired style.
+The existing v0's `Divider` control supports a very similar set of props to the one being proposed for v9 with a few differences that are outlined below:
 
-The `fitted` property also is not in use as by default the `Divider` can use the standard `margin` | `marginLeft` | `marginRight` | `marginTop` | `marginBottom` properties to adjust as needed. The default margin for the divider is 0.
+- `content` => NOT SUPPORTED - use `children` instead
```
Comment (behowell) [inline packages/react-components/react-divider/MIGRATION.md:17]: Ditto to above:
```suggestion
- `content` => children of the `Divider`
```

File: packages/react-components/react-divider/README.md [hunk 6]
```diff
@@ -1,23 +1,39 @@
 # @fluentui/react-divider
 
-**React Divider components for [Fluent UI](https://dev.microsoft.com/fluentui)**
+**Divider components for [Fluent UI](https://aka.ms/fluentui-storybook)**
```
Comment (behowell) [inline packages/react-components/react-divider/README.md:3]: Should this URL be updated to https://react.fluentui.dev/? (Maybe in a separate PR that updates all READMEs). Or maybe we can leave the aka.ms link here, and update that to redirect to the new URL.

**Review by behowell — APPROVED** [review]

File: packages/react-components/react-divider/MIGRATION.md (hunk 3 above)
Comment (khmakoto) [reply packages/react-components/react-divider/MIGRATION.md:9]: `style` is a prop, not `styles`, right?

File: packages/react-components/react-divider/README.md (hunk 6 above)
Comment (khmakoto) [reply packages/react-components/react-divider/README.md:3]: I'll follow up with this in a separate PR.
